cypress/e2e/CMS/$HomeAndLandingPageTestCreationOfContent/index.js

```diff
@@ -0,0 +1,215 @@
+// <reference types="Cypress" />
+import { And } from 'cypress-cucumber-preprocessor/steps';
+function createRandomStr() {
+    var result = '';
+    var characters = 'abcdefghijklmnopqrstuvwxyz0123456789';
+    var charactersLength = characters.length;
+    for (var i = 0; i < 5; i++) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    }
+    return result;
+}
+let randomNum = createRandomStr();
+const siteSection = Cypress.env('test_site_section');
+const frontEndBaseUrl = Cypress.env('front_end_base_url');
+
+And('user fills out the following fields', (dataTable) => {
+    for (const { fieldLabel, value, field_name } of dataTable.hashes()) {
+        if (fieldLabel === 'Pretty URL') {
+            value = `${value}-${randomNum}`;
+        }
+        cy.get(`input[name^='${field_name}']`).as('inputField').parent().find('label').should('include.text', fieldLabel);
+        cy.get('@inputField').type(value);
+    }
+});
+
+And('user clicks on title with url {string} from the list of content', (contentHref) => {
+    cy.get(`a[href='${siteSection}/${contentHref}-${randomNum}']`).click();
+});
+
+And('user selects a checkbox next to title with url {string} from the list of content', (url) => {
+    cy.get(`a[href='${siteSection}/${url}-${randomNum}']`).parent().parent().find('input.form-checkbox').check();
+});
+
+Given('user is navigating to the front end site with path site section plus {string}', (purl) => {
+    cy.visit(`${frontEndBaseUrl}${siteSection}/${purl}-${randomNum}`, { retryOnStatusCodeFailure: true });
+});
+
+And('the content item with url {string} does not exist in the list of content', (url) => {
+    cy.get(`a[href='${siteSection}/${url}-${randomNum}']`).should('not.exist');
+});
+
+
+/*--------Scenario: User is adding new Home and Landing page content type and adding a Primary Feature Card Row -------*/
+And('user selects {string} from Contents dropdown',(dropDown) =>{
```

Comment:
steps at line 44 and at line 56 are identical - since the locators and logic is the same, can we combine them into one to reduce duplication?

Reply:
if you prefer to keep them separate, then I want to see more specific locators to ensure it indeed interacts with a correct element